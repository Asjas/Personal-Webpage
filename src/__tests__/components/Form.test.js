import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import ContactForm from '../../components/Form';

describe('Contact form', () => {
  it('renders correctly', () => {
    const { container } = render(<ContactForm />);

    expect(container).toMatchSnapshot();
  });

  it('updates the different input fields', () => {
    const name = 'A-J Roos';
    const email = 'asjas@outlook.com';
    const message = 'This is a test message';

    const { getByLabelText } = render(<ContactForm />);
    const nameInput = getByLabelText('Enter your name:');
    const emailInput = getByLabelText('Enter your email:');
    const messageInput = getByLabelText('Enter a message:');

    nameInput.textContent = name;
    emailInput.textContent = email;
    messageInput.textContent = message;

    expect(nameInput).toHaveTextContent(name);
    expect(emailInput).toHaveTextContent(email);
    expect(messageInput).toHaveTextContent(message);
  });

  it('is disabled when the user submits the form', () => {
    const { getByText, getByTestId } = render(<ContactForm />);

    fireEvent.click(getByText('Submit'));
    const { disabled } = getByTestId('form-disabled');

    expect(disabled).toBe(true);
  });
});
