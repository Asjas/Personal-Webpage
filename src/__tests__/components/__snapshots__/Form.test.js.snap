// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact form renders correctly and matches snapshot 1`] = `
<div>
  <form>
    <fieldset
      aria-busy="false"
      data-testid="form-disabled"
    >
      <label
        for="name"
      >
        <span>
          Enter your name:
        </span>
        <span
          component="span"
          name="name"
        />
        <input
          id="name"
          name="name"
          placeholder="full name"
          type="text"
        />
      </label>
      <label
        for="email"
      >
        <span>
          Enter your email:
        </span>
        <span
          component="span"
          name="email"
        />
        <input
          id="email"
          name="email"
          placeholder="example@gmail.com"
          type="email"
        />
      </label>
      <label
        for="message"
      >
        <span>
          Enter a message:
        </span>
        <span
          component="span"
          name="message"
        />
        <textarea
          component="textarea"
          id="message"
          name="message"
          placeholder="type your message here"
        />
      </label>
      <button
        class="button button__primary"
        type="submit"
      >
        Submit
      </button>
    </fieldset>
  </form>
</div>
`;
